### PR TITLE
Added full RP2040 (RPi Pico) support 

### PR DESCRIPTION
**Pull request by Intubun**
Added the the Option to choose the 2nd SPI Bus on the RP2040 with the RP2040. Added also the specific commands to change the Bus pins on the RP2040. Added some documentation as well. The Addition is intended for the arduino-pico library by earlephilhower https://github.com/earlephilhower/arduino-pico

(At the moment the edits are validated on the original Pico board and the MCP4822)

### CODE REVIEW

**Comment by Intubun** [comment]
Btw. I also added the current version in the pull request for a possible release...

**Comment by RobTillaart** [comment]
Thanks for the addition
Will check it later this weekend. 
(Might be usefull for other SPI based libs too)

**Comment by Intubun** [comment]
Yeah, I think so too... I also (if this one will pass) am planning to add this capability to some other libs, which I need for a upcoming project...

File: MCP_DAC.cpp [hunk 1]
```diff
@@ -70,6 +70,19 @@ void MCP_DAC::begin(uint8_t select)
       mySPI->end();
       mySPI->begin(18, 19, 23, select);   // CLK=18 MISO=19 MOSI=23
     }
+  
+    #elif defined(ARDUINO_ARCH_RP2040)
+    
+    if (_useSPI == false &&  _useSPI1 == true) {
```
Comment (RobTillaart) [inline MCP_DAC.cpp]: prefer extra brackets so evaluation is 100% clear.
 if ((_useSPI == false) && (_useSPI1 == true))

File: MCP_DAC.h [hunk 2]
```diff
@@ -74,47 +74,78 @@ class MCP_DAC
...
 
-  // to overrule ESP32 default hardware pins
+  // to overrule the ESP32s default hardware pins
+  void     setGPIOpins(uint8_t clk, uint8_t miso, uint8_t mosi, uint8_t select);
+
+  
+  #elif defined(ARDUINO_ARCH_RP2040)    // RP2040 specific
+
+  // check which SPI-Bus (SPI or SPI1) is used
+  void     selectSPI()  { _useSPI = true;   _useSPI1 = false;  };
```
Comment (RobTillaart) [inline MCP_DAC.h]: One variable should be enough to indicate which SPI is used?
(compare the ESP32   _useHSPI  flag

File: examples/MCP4921_wave_generator_RP2040/.arduino-ci.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  platforms:
+    # - uno
+    # - due
+    # - zero
+    # - leonardo
+    # - m4
+    # - esp32
+    # - esp8266
+    # - mega2560
+    - ARDUINO_ARCH_RP2040
```
Comment (RobTillaart) [inline examples/MCP4921_wave_generator_RP2040/.arduino-ci.yml]: RP2040 is not supported in the Arduino build-CI
So you should comment that line (all lines in fact)

Created an issue here - https://github.com/Arduino-CI/arduino_ci/issues/329

**Comment by RobTillaart** [comment]
- If you comment the RP2040 from the - .arduino-ci.yml file at the  specific example, build will succeed
- I will update all the version strings etc in the lib / meta files after the merge of this one
  - .cpp, library.json, library.properties.

**Comment by RobTillaart** [comment]
Had a quick look, and I could not find any RPI pico / RP2040 workflow to test the library.
I you find one it should be added to the workflows folder.

**Comment by RobTillaart** [comment]
@Intubun 

Found how to add a platform to the build-CI. 

Edit **.arduino-ci.yml**

Had to install the boards support for the pico  (packages part) - https://github.com/earlephilhower/arduino-pico
Found the board and package definition in the boards.txt file of the installed package
The define might not be needed..

```
platforms:
  rpipico:
    board: rp2040:rp2040:rpipico
    package: rp2040:rp2040
    gcc:
      features:
      defines:
        - ARDUINO_ARCH_RP2040
      warnings:
      flags:

packages:
  rp2040:rp2040:
    url: https://github.com/earlephilhower/arduino-pico/releases/download/global/package_rp2040_index.json


compile:
  # Choosing to run compilation tests on 2 different Arduino platforms
  platforms:
    - uno
    # - due
    # - zero
    # - leonardo
    # - m4
    # - esp32
    # - esp8266
    # - mega2560
    - rpipico
```

- https://github.com/Arduino-CI/arduino_ci
  - how to edit the yml file
- https://github.com/arduino/Arduino/wiki/Unofficial-list-of-3rd-party-boards-support-urls
  - https://github.com/earlephilhower/arduino-pico/releases/download/global/package_rp2040_index.json

**Comment by Intubun** [comment]
So, it was only an Issue with the check not knowing the RP2040, right?

**Comment by Intubun** [comment]
Soo... Ive changed the code slightly... Should work now... It uses now only one variable for the bus selection, like the ESP32...

**Comment by Intubun** [comment]
Sorry forgot to change the .arduino-ci.yml for the test...

**Comment by RobTillaart** [comment]
> So, it was only an Issue with the check not knowing the RP2040, right?

Yes, the changes themselves looked good (few minor things).

Now I know how to add RP2040 testing I plan to add it to all my other 150++ libraries, see if they are portable.
(will take some time..)

> ... add this capability to some other libs, which I need for a upcoming project...

Can you tell which libraries? 
And what your project is about? (if it is confidential, I understand)

**Comment by RobTillaart** [comment]
Build succeeds 😎 well done!

**Pull request merged**